unit_tests/test_dbprocessing.py

```diff
@@ -25,23 +27,37 @@ class ProcessQueueTestsBase(unittest.TestCase, dbp_testing.AddtoDBMixin):
     def setUp(self):
         """Make a db and open it so have something to work with"""
         super(ProcessQueueTestsBase, self).setUp()
-        self.td = tempfile.mkdtemp()
-        shutil.copy2(
-            os.path.join(os.path.dirname(__file__), 'emptyDB.sqlite'),
-            self.td)
+
+        if 'PGDATABASE' in os.environ.keys():
+            self.td = os.environ['PGDATABASE']
+        else:
+            self.td = tempfile.mkdtemp()
+            shutil.copy2(
+                os.path.join(os.path.dirname(__file__), 'emptyDB.sqlite'),
+                self.td)
+
         # Set up the baseline mission environment, BEFORE making processqueue
         self.addSkeletonMission()
-        self.pq = dbprocessing.dbprocessing.ProcessQueue(
-            os.path.join(self.td, 'emptyDB.sqlite'))
+        if 'PGDATABASE' in os.environ.keys():
+            self.pq = dbprocessing.dbprocessing.ProcessQueue(self.td)
+        else:
+            self.pq = dbprocessing.dbprocessing.ProcessQueue(
+                    os.path.join(self.td, 'emptyDB.sqlite'))
         self.dbu = self.pq.dbu
 
     def tearDown(self):
         """Remove the db and working tree"""
-        self.dbu.closeDB() # Before the database is removed...
-        del self.dbu
         # Unfortunately all the cleanup is in the destructor
-        del self.pq
-        shutil.rmtree(self.td)
+        #del self.pq
+        #shutil.rmtree(self.td)
+
+        if 'PGDATABASE' in os.environ.keys():
+            for table in reversed(self.dbu.metadata.sorted_tables):
+                self.dbu.session.execute(table.delete());
+                self.dbu.session.commit()
+
+            self.dbu.closeDB() # Before the database is removed...
+        del self.dbu
```

Comment:
This is going to cause serious problems in sqlite because the dbu isn't closed and the temp directory is never deleted, and in either case the process queue isn't cleaned up before taking out all the tables. What's the thought here?

Reply:
Not sure why github does not expand, it's hard to see the whole code in these reviews.

Reply:
Looking further, that code is not done.  I don't have a SQL section.

Reply:
So if this isn't done we should maybe take it out before this merge?

Reply:
Ah, right, delete clears tables, doesn't drop them. Reshuffling so that all the cleanup still happens.